src/test/java/mServer/tool/M3U8UtilsTest.java

```diff
@@ -0,0 +1,92 @@
+    public static final String AWAITED_URL_HD = "http://ondemand-ww.wdr.de/medp/fsk0/131/1315091/1315091_15301369.mp4";
+
+    /**
+     * Tests {@link M3U8Utils#gatherUrlsFromWDRM3U8(String)} with a URL with more then three qualities.
+     */
+    @Test
+    public void testGatherUrlsFromWDRM3U8_MoreThenThree_Positiv()
+    {
+        Map<Qualities, String> qualitiesAndUrls = M3U8Utils.gatherUrlsFromWDRM3U8(TEST_URL_POSITIV_MORE_THEN_THREE_ELEMENTS);
+        Assert.assertThat(qualitiesAndUrls.get(Qualities.SMALL), CoreMatchers.is(AWAITED_URL_SMALL));
+        Assert.assertThat(qualitiesAndUrls.get(Qualities.NORMAL), CoreMatchers.is(AWAITED_URL_NORMAL));
+        Assert.assertThat(qualitiesAndUrls.get(Qualities.HD), CoreMatchers.is(AWAITED_URL_HD));
+    }
+
+    /**
+     * Tests {@link M3U8Utils#gatherUrlsFromWDRM3U8(String)} with a URL with three qualities.
+     */
+    @Test
+    public void testGatherUrlsFromWDRM3U8_ExactThree_Positiv()
+    {
+        Map<Qualities, String> qualitiesAndUrls = M3U8Utils.gatherUrlsFromWDRM3U8(TEST_URL_POSITIV_THREE_ELEMENTS);
+        Assert.assertThat(qualitiesAndUrls.get(Qualities.SMALL), CoreMatchers.is(AWAITED_URL_SMALL));
+        Assert.assertThat(qualitiesAndUrls.get(Qualities.NORMAL), CoreMatchers.is(AWAITED_URL_NORMAL));
+        Assert.assertThat(qualitiesAndUrls.get(Qualities.HD), CoreMatchers.is(AWAITED_URL_HD));
+    }
+
+    /**
+     * Tests {@link M3U8Utils#gatherUrlsFromWDRM3U8(String)} with a URL with only two qualities.
+     */
+    @Test
+    public void testGatherUrlsFromWDRM3U8_ExactTwo_Positiv()
+    {
+        Map<Qualities, String> qualitiesAndUrls = M3U8Utils.gatherUrlsFromWDRM3U8(TEST_URL_POSITIV_TWO_ELEMENTS);
+        Assert.assertThat(qualitiesAndUrls.get(Qualities.SMALL), CoreMatchers.is(AWAITED_URL_SMALL));
+        Assert.assertThat(qualitiesAndUrls.get(Qualities.NORMAL), CoreMatchers.is(AWAITED_URL_NORMAL));
+    }
+
+    /**
+     * Tests {@link M3U8Utils#gatherUrlsFromWDRM3U8(String)} with a URL with only one quality.
+     */
+    @Test
+    public void testGatherUrlsFromWDRM3U8_ExactOne_Positiv()
+    {
+        Map<Qualities, String> qualitiesAndUrls = M3U8Utils.gatherUrlsFromWDRM3U8(TEST_URL_POSITIV_ONE_ELEMENT);
+        Assert.assertThat(qualitiesAndUrls.get(Qualities.SMALL), CoreMatchers.is(AWAITED_URL_SMALL));
+    }
+
+    /**
+     * Tests {@link M3U8Utils#gatherUrlsFromWDRM3U8(String)} with a URL without a quality.
+     */
+    @Test
+    public void testGatherUrlsFromWDRM3U8_NoneElement_Negativ()
+    {
+        Map<Qualities, String> qualitiesAndUrls = M3U8Utils.gatherUrlsFromWDRM3U8(TEST_URL_NEGATIV_NONE_ELEMENT);
+        Assert.assertThat(qualitiesAndUrls.isEmpty(), CoreMatchers.is(true));
+    }
+
+    /**
+     * Tests {@link M3U8Utils#gatherUrlsFromWDRM3U8(String)} with a wrong URL.
+     */
+    @Test
+    public void testGatherUrlsFromWDRM3U8_WrongUrl_Negativ()
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00100)